usaspending_api/database_scripts/etl/award_delta_view.sql

```diff
@@ -71,7 +71,7 @@ SELECT
       THEN CONCAT(
         '{"name":"', vw_award_search.funding_toptier_agency_name,
         '","code":"', vw_award_search.funding_toptier_agency_code,
-        '","id":"', vw_award_search.funding_agency_id, '"}'
+        '","id":"', (SELECT aa.id FROM Agency aa WHERE aa.toptier_agency_id = (SELECT aaa.toptier_agency_id FROM Agency aaa WHERE aaa.id = vw_award_search.funding_agency_id) ORDER BY aa.toptier_flag DESC, aa.id LIMIT 1), '"}'
```

Comment:
Wow.  That is some fantastic SQL.  Good job.  👍

Reply:
Although `Agency` with a capital `A` is messing with my OCD.

Reply:
It dawns on me that another way to do this would be to resolve all of the agencies in a subquery or cte so that they  don't have to be correlated subqueries.  Might help with performance.  Not sure what performance is like on this but the subqueries couldn't have helped.

Reply:
![Alt Text](https://media.giphy.com/media/l4uUJbflyA3Aow8hYf/giphy.gif)

Reply:
(in case anyone is confused, that is Kirk ^^^^^^^^^^^^)

Reply:
No one was confused, they've just forgotten what I look like.